maven-core/src/main/java/org/apache/maven/lifecycle/internal/MojoExecutor.java

```diff
@@ -232,7 +237,17 @@ private static class ProjectLock implements AutoCloseable
                 boolean aggregator = mojoDescriptor.isAggregator();
                 acquiredAggregatorLock = aggregator ? aggregatorLock.writeLock() : aggregatorLock.readLock();
                 acquiredProjectLock = getProjectLock( session );
-                acquiredAggregatorLock.lock();
+                if ( !acquiredAggregatorLock.tryLock() )
+                {
+                    for ( String s : MessageHelper.formatWarning(
+                            "An aggregator Mojo is already executing in parallel build, but aggregator "
+                                    + "Mojos require exclusive access to reactor to prevent race conditions. This "
+                                    + "mojo execution will be blocked until the aggregator work is done." ) )
+                    {
```

Comment:
I have commented on this already somewhere else. Can you find that up? I can't :-(

Reply:
Found it.  But that makes me wonder if we should display some more detailed infos about the aggregator mojo.  It may not be trivial to find it in the whole build.  Thoughts ?

Reply:
Yes, context info would be good.